src/freenet/client/filter/M3UFilter.java

```diff
@@ -0,0 +1,171 @@
+                            final byte utf80 = (byte)'0';
+                            final byte utf89 = (byte)'9';
+                            final byte utf8A = (byte)'A';
+                            final byte utf8Z = (byte)'Z';
+                            final byte utf8a = (byte)'a';
+                            final byte utf8z = (byte)'z';
+                            final byte utf8dot = (byte)'.';
+                            final byte utf8dash = (byte)'-';
+                            for (int i = 0; i < fileIndex; i++) {
+                                byte b = fileUri[i];
+                                if (!((utf80 <= b && b <= utf89) ||
+                                      (utf8A <= b && b <= utf8Z) ||
+                                      (utf8a <= b && b <= utf8z) ||
+                                      utf8dot == b ||
+                                      utf8dash == b)) {
+                                    isBadUri = true;
+                                    break;
+                                } 
+                                if ((byte)'.' == b) {
+                                    numberOfDotsInUri += 1;
+                                }
+                            }
+                            if (numberOfDotsInUri != 1) {
+                                isBadUri = true;
+                            }
+                            // use only the first fileIndex bytes
+                            String uri;
+                            // TODO: cleanly filter the URI, i.e. with processURI from GenericReadFilterCallback.
+                            if (isBadUri) {
+                                uri = badUriReplacement;
+                            } else {
+                                uri = new String(fileUri, 0, fileIndex, "UTF-8");
+                            }
+                            output.write(uri.getBytes("UTF-8"));
+                            output.write(nextchar);
+                        }
+                    }
+                    // skip the newline
+                    readcount = dis.read(nextchar);
+                    break;
+                }
+            }
+        }
+		output.flush();
+	}
+
+	private static String l10n(String key) {
+		return NodeL10n.getBase().getString("M3UFilter."+key);
+	}
+
+	private void throwHeaderError(String shortReason, String reason) throws DataFilterException {
+		// Throw an exception
+		String message = l10n("notGif");
+		if(reason != null) message += ' ' + reason;
+		if(shortReason != null)
+			message += " - (" + shortReason + ')';
+		throw new DataFilterException(shortReason, shortReason, message);
+	}
+
+	@Override
+	public void writeFilter(InputStream input, OutputStream output, String charset, HashMap<String, String> otherParams,
+	        FilterCallback cb) throws DataFilterException, IOException {
+		output.write(input.read());
+		return;
```

Comment:
that’s just copied from what the gif filter did… I guess now I have two files to correct :)